public/app/features/dashboard/dashgrid/AddPanelPanel.tsx

```diff
@@ -101,19 +119,85 @@ export class AddPanelPanel extends React.Component<AddPanelPanelProps, AddPanelP
     );
   }
 
+  filterChange(evt) {
+    this.setState({ filter: evt.target.value });
+    this.setState({ panelPlugins: this.getPanelPlugins(evt.target.value) });
```

Comment:
you do not need to call setState multiple times, you can pass in everything in one call